ios/FluentUI/Bottom Commanding/CommandingItem.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// An object representing a command.
+///
+/// `CommandingItem` defines the high level properties and behavior of a command. Its visual representation is determined by
+/// the `BottomCommandingController`.
+@objc(MSFCommandingItem)
+open class CommandingItem: NSObject {
+
+    /// A closure that's called when the command is triggered
+    @objc open var action: (CommandingItem) -> Void
+
+    /// The title of the command item.
+    @objc open var title: String {
+        didSet {
+            if title != oldValue {
+                delegate?.commandingItem(self, didChangeTitleTo: title)
+            }
+        }
+    }
+
+    /// A `UIImage` to be displayed with the command.
+    @objc open var image: UIImage {
+        didSet {
+            if image != oldValue {
+                delegate?.commandingItem(self, didChangeImageTo: image)
+            }
+        }
+    }
+
+    /// A `UIImage` used when the command is represented as a button in selected state.
+    @objc open var selectedImage: UIImage? {
+        didSet {
+            if selectedImage != oldValue {
+                delegate?.commandingItem(self, didChangeSelectedImageTo: selectedImage)
+            }
+        }
+    }
+
+    /// Used in large content viewer if this command is represented using a view that cannot scale with Dynamic Type.
+    ///
+    /// When this is `nil`, `image` will be used instead.
+    @objc open var largeImage: UIImage? {
+        didSet {
+            if largeImage != oldValue {
+                delegate?.commandingItem(self, didChangeLargeImageTo: largeImage)
+            }
+        }
+    }
+
+    /// Indicates whether the command is currently on.
+    ///
+    /// When `isToggleable` is `true`, this property is toggled automatically before `action` is called.
+    @objc open var isOn: Bool {
+        didSet {
+            if isOn != oldValue {
+                delegate?.commandingItem(self, didChangeOnTo: isOn)
+            }
+        }
+    }
+
+    /// Indicates whether the command is enabled.
+    @objc open var isEnabled: Bool {
+        didSet {
+            if isEnabled != oldValue {
+                delegate?.commandingItem(self, didChangeEnabledTo: isEnabled)
+            }
+        }
+    }
+
+    /// Indicates whether `isOn` should be toggled automatically before `action` is called.
+    @objc open var isToggleable: Bool {
```

Comment:
this one only make sense if it not used for heroitem right and used for boolean cell? should we note that?

Reply:
No it will work as described for both hero items and boolean cells.